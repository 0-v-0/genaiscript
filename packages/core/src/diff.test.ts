import { expect, test } from 'vitest'
import { parseLLMDiffs } from "./diff"

test("is_valid_email", () => {
    const source = 
`[1] import re
[2] 
[3] def is_valid_email(email):
- [4]     if re.fullmatch(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+", email):
+ [4]     pattern = re.compile(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+")
+ [5]     if pattern.fullmatch(email):
[6]         return True
[7]     else:
[8]         return False`    
    const chunks = parseLLMDiffs(source)
    console.log(chunks)
    expect(chunks.length).toBe(4)
    expect(chunks).toMatchSnapshot()
})