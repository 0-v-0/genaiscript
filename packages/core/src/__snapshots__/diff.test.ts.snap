// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`diff 1`] = `
[
  {
    "lineNumbers": [
      1,
      2,
      3,
    ],
    "lines": [
      "import re",
      "",
      "def is_valid_email(email):",
    ],
    "state": "existing",
  },
  {
    "lineNumbers": [
      4,
    ],
    "lines": [
      "    if re.fullmatch(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+", email):",
    ],
    "state": "deleted",
  },
  {
    "lineNumbers": [
      4,
      5,
    ],
    "lines": [
      "    pattern = re.compile(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+")",
      "    if pattern.fullmatch(email):",
    ],
    "state": "added",
  },
  {
    "lineNumbers": [
      6,
      7,
      8,
    ],
    "lines": [
      "        return True",
      "    else:",
      "        return False",
    ],
    "state": "existing",
  },
]
`;

exports[`is_valid_email 1`] = `
[
  {
    "lineNumbers": [
      1,
      2,
      3,
    ],
    "lines": [
      "import re",
      "",
      "def is_valid_email(email):",
    ],
    "state": "existing",
  },
  {
    "lineNumbers": [
      4,
    ],
    "lines": [
      "    if re.fullmatch(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+", email):",
    ],
    "state": "deleted",
  },
  {
    "lineNumbers": [
      4,
      5,
    ],
    "lines": [
      "    pattern = re.compile(r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+")",
      "    if pattern.fullmatch(email):",
    ],
    "state": "added",
  },
  {
    "lineNumbers": [
      6,
      7,
      8,
    ],
    "lines": [
      "        return True",
      "    else:",
      "        return False",
    ],
    "state": "existing",
  },
]
`;
